core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,91 @@
-require('domain/opportunity/SkillOpportunityObjectFactory.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  UrlInterpolationService
+} from 'domain/utilities/url-interpolation.service';
+import {
+  ExplorationOpportunitySummary
+} from 'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+import {
+  SkillOpportunity
+} from 'domain/opportunity/SkillOpportunityObjectFactory';
 
-angular.module('oppia').factory('ContributionOpportunitiesBackendApiService', [
-  '$http', 'ExplorationOpportunitySummaryObjectFactory',
-  'SkillOpportunityObjectFactory', 'UrlInterpolationService',
-  'OPPORTUNITY_TYPE_SKILL', 'OPPORTUNITY_TYPE_TRANSLATION',
-  'OPPORTUNITY_TYPE_VOICEOVER',
-  function($http, ExplorationOpportunitySummaryObjectFactory,
-      SkillOpportunityObjectFactory, UrlInterpolationService,
-      OPPORTUNITY_TYPE_SKILL, OPPORTUNITY_TYPE_TRANSLATION,
-      OPPORTUNITY_TYPE_VOICEOVER) {
-    var urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
-    var _getOpportunityFromDict = function(opportunityType, opportunityDict) {
-      if (
-        opportunityType === OPPORTUNITY_TYPE_VOICEOVER ||
-        opportunityType === OPPORTUNITY_TYPE_TRANSLATION) {
-        return ExplorationOpportunitySummaryObjectFactory.createFromBackendDict(
-          opportunityDict);
-      } else if (opportunityType === OPPORTUNITY_TYPE_SKILL) {
-        return SkillOpportunityObjectFactory.createFromBackendDict(
-          opportunityDict);
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesBackendApiService {
+  urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
+  /**  TODO(srijanreddy98): for some reason not able to import types (
+  * OPPORTUNITY_TYPE_SKILL,
+  * OPPORTUNITY_TYPE_TRANSLATION,
+  * OPPORTUNITY_TYPE_VOICEOVER
+  * )from constants.ts in assets */
+  _getOpportunityFromDict(opportunityType, opportunityDict) {
+    if (
+      opportunityType === 'voiceover' ||
+      opportunityType === 'translation') {
+      return new ExplorationOpportunitySummary(opportunityDict.id,
+        opportunityDict.topic_name, opportunityDict.story_title,
+        opportunityDict.chapter_title, opportunityDict.content_count,
+        opportunityDict.translation_counts);
+    } else if (opportunityType === 'skill') {
+      return new SkillOpportunity(
+        opportunityDict.id, opportunityDict.skill_description,
+        opportunityDict.topic_name, opportunityDict.question_count);
+    }
+  }
+
+  _fetchOpportunities(opportunityType, params, successCallback) {
+    this.http.get(this.urlInterpolationService.interpolateUrl(
+      this.urlTemplate, { opportunityType }
+    ), {
+      params
+    }).toPromise().then((data: any) => {
```

Comment:
Move this to the above line? `    ), { params }).toPromise().then((data: any) => {`


/cc @bansalnitish, do we allow using `property value shorthand` for dict?

Reply:
Done